desktop/core/src/desktop/models.py

```diff
@@ -1068,9 +1068,23 @@ def get_home_directory(self, user):
 
   def get_gist_directory(self, user):
     home_dir = self.get_home_directory(user)
-    gist_dir, created = Directory.objects.get_or_create(name=Document2.GIST_DIR, owner=user, parent_directory=home_dir)
-    if created:
-      LOG.info('Successfully created gist directory for user: %s' % user.username)
+
+    try:
+      gist_dir, created = Directory.objects.get_or_create(name=Document2.GIST_DIR, owner=user, parent_directory=home_dir)
+      if created:
+        LOG.info('Successfully created gist directory for user: %s' % user.username)
+    except Directory.MultipleObjectsReturned:
+      LOG.exception('Multiple Gist directories detected. Merging all into one.')
+
+      all_gist_dirs = self.filter(owner=user, parent_directory=home_dir, name=Document2.GIST_DIR, type='directory').order_by('-last_modified')
+      gist_dir = all_gist_dirs.last()
+      gist_dirs_dup = all_gist_dirs.exclude(uuid=gist_dir.uuid)
+
+      for dir in gist_dirs_dup:
+        dir.children.exclude(name='.Trash').update(parent_directory=gist_dir)
```

Comment:
Tried using `prefetch_related('children')` to fetch all children gists from gist_dup_dirs not able to get a way to use `exclude(name='.Trash')` @romainr 
Shall we check for it like this 

for child in gist_dirs_dup_children:
      if not child.name == '.Trash':
        update parent_dir of children

Reply:
Let's go with it, KISS is good

just .Trask --> Document2.TRASH_DIR constant

But actually, I don't see why we need to care about some trash folders inside the dup gist dirs?
Also add a test for that?